.github/workflows/CD.yml

```
@@ -0,0 +1,21 @@
+name: CD 
+
+on:
+  workflow_run:
+    workflows: ["Rust CI Pipeline"]  # Trigger on successful CI completion
+    types:
+      - completed
```

A successful CI pipeline is still not sufficient to deploy anything. CD should be triggered when a PR is merged, which fires a push event to main.

```suggestion
on:
  push:
    branches: [main]
```